grafana-plugin/src/pages/schedules/Schedules.tsx

```diff
@@ -395,17 +412,21 @@ class SchedulesPage extends React.Component<SchedulesPageProps, SchedulesPageSta
   };
 
   handleSchedulesFiltersChange = (filters: SchedulesFiltersType) => {
-    this.setState({ filters }, this.debouncedUpdateSchedules);
+    this.setState({ filters, isLoading: true }, () => this.debouncedUpdateSchedules(filters));
   };
 
-  applyFilters = () => {
-    const { filters } = this.state;
-    const { store } = this.props;
-    const { scheduleStore } = store;
-    scheduleStore.updateItems(filters);
+  applyFilters = (filters: SchedulesFiltersType) => {
+    const { scheduleStore } = this.props.store;
+
+    scheduleStore.updateItems(filters).finally(() => {
+      // check for any change in filters in the meanwhile
+      if (filters === this.state.filters) {
```

Comment:
but filters object is not a primitive, how do you compare them with just `===`

Reply:
This checks for reference as that's enough to know a change happened. It's overhead to check for actual change and doesn't bring any benefit I'd say.

This check is only so that we do not reset `isLoading` to false when there's like 2 requests ongoing.